lib/dynflow/watchers/memory_consumption_watcher.rb

```diff
@@ -0,0 +1,32 @@
+require 'get_process_mem'
+
+module Dynflow
+  module Watchers
+    class MemoryConsumptionWatcher
+
+      attr_reader :memory_limit, :world
+
+      def initialize(world, memory_limit, options)
+        @memory_limit = memory_limit
+        @world = world
+        @polling_interval = options[:polling_interval] || 10
+        @memory_watcher = options[:memory_watcher] || GetProcessMem.new
```

Comment:
`s/memory_watcher/memory_info_loader` or somethng similar, the watcher is this class